commit c2cd650b5b2bd6b57f2f62be65b1a8a576d4de6d
Author: Antoine Tenart <atenart@kernel.org>
Date:   Fri Jun 22 11:50:52 2018 +0200

    net: mscc: make sparse happy
    
    This patch fixes a sparse warning about using an incorrect type in
    argument 2 of ocelot_write_rix(), as an u32 was expected but a __be32
    was given. The conversion to u32 is forced, which is safe as the value
    will be written as-is in the hardware without any modification.
    
    Fixes: 08d02364b12f ("net: mscc: fix the injection header")
    Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index 52c57e0ff617..776a8a9be8e3 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -374,7 +374,8 @@ static int ocelot_port_xmit(struct sk_buff *skb, struct net_device *dev)
 	ocelot_gen_ifh(ifh, &info);
 
 	for (i = 0; i < IFH_LEN; i++)
-		ocelot_write_rix(ocelot, cpu_to_be32(ifh[i]), QS_INJ_WR, grp);
+		ocelot_write_rix(ocelot, (__force u32)cpu_to_be32(ifh[i]),
+				 QS_INJ_WR, grp);
 
 	count = (skb->len + 3) / 4;
 	last = skb->len % 4;